File: 0027-remove-element/0027-remove-element.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int start = 0, end = nums.length-1;  
        for(int i = end; i >= 0; i--){
            if(nums[i] != val){
                end = i; 
                break;  
            }
        }

        //System.out.println(end); 

        while(start < end){
            if(nums[start] == val){
                int tmp = nums[start]; 
                nums[start] = nums[end]; 
                nums[end] = tmp; 
                //start++;
                end--; 
            }
            start++; 
        }

        //System.out.print(Arrays.toString(nums)); 

        return start + 1;  
    }
}

File: 0027-remove-element/0027-remove-element.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int index = 0; 
        for(int i = 0; i < nums.size(); i++){
            if(nums[i] != val){
                swap(nums[i], nums[index++]); 
            }
        }
        return index; 
    }
};